java/ql/src/semmle/code/java/frameworks/apache/Lang.qll

```diff
@@ -63,3 +63,64 @@ private class ApacheLangArrayUtilsTaintPreservingMethod extends TaintPreservingC
     src = [0, 2]
   }
 }
+
+private Type getAnExcludedParameterType() {
+  result instanceof PrimitiveType or
+  result.(RefType).hasQualifiedName("java.nio.charset", "Charset") or
+  result.(RefType).hasQualifiedName("java.util", "Locale")
+}
+
+private class SafeJoinArrayType extends PrimitiveType {
+  SafeJoinArrayType() { getName().regexpMatch("byte|double|float|int|long|short") }
+}
```

Comment:
```suggestion
```

Reply:
Doh. Mind approving so I can merge once this is done?

Reply:
Can't approve just yet, as there are still changes I expect (and any push will remove approval). Also, could you please keep the review fixes in their own commits? Force-pushes make review difficult. (Force-pushing when rebasing is fine, though).

Reply:
OK will do in future. This change is pushed already.

Reply:
Protip btw, you can click the words "force-pushed" to see the diff before vs. after the push

Reply:
I know, but in my experience that doesn't always work very well. It seems like what you get is somewhat ill-defined and a best-effort approach at most.

Reply:
E.g. clicking on the second-to-last "force-pushed" shows me several unrelated changes, possibly related to a rebase, but if that's then mixed with actual changes then those parts are very hard to separate.